### DISCUSSION TITLE
I want to create a folder using GitHub API

### DISCUSSION THREAD

**Discussion by ryoshindo** [community/community#24518] [Apps, API and Webhooks]
<p>I’ve recently created a GitHub organization and have been working on building the first environment and so on.</p>
<p>I was able to create a new repository in a GitHub organization using GitHub API, but I’m having trouble creating a folder under the repository using API.</p>
<p>I’d like to know if anyone can tell me more about this.<br>
Thank you for your help.</p>

**Comment by ryoshindo** [comment]
<p>I was able to create a folder from following URL.<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <a href="https://developer.github.com/v3/repos/contents/" target="_blank" rel="nofollow noopener">GitHub Developer</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/59394380/181085157-e1c519bb-2344-42e8-af60-6b667d5e5522.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_10x10.png">

<h3><a href="https://developer.github.com/v3/repos/contents/" target="_blank" rel="nofollow noopener">Contents</a></h3>

<p>Get started with one of our guides, or jump straight into the API documentation.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Thank you for joining the community <a class="mention" href="/ryoshindo">@ryoshindo</a>, I see you’ve found our API documentation. Here is some additional reading that you might find useful:</p>
<ul>
<li>Adding a file, reference from <code>octokit.rest</code> <a href="https://octokit.rest/PUT/repos/%7Bowner%7D/%7Brepo%7D/contents/%7Bpath%7D">https://octokit.rest/PUT/repos/{owner}/{repo}/contents/{path}</a>
</li>
</ul>

**Comment by ryoshindo** [comment]
<p>Octokit is very useful!<br>
Thank you for your support.</p>

**Reply by Karan-S-dev** [reply]
Hi,

I tried using octokit but its showing

{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#create-or-update-file-contents"
}

**Reply by swansong314** [reply]
Sorry for the necro, but I also ran into this issue. I was using a PAT to talk to the GitHub API to create a workflow file. I hit this error due to incorrect permissions on the PAT:
> OAuth app tokens and personal access tokens (classic) need the repo scope to use this endpoint. The workflow scope is also required in order to modify files in the .github/workflows directory.

from: https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents

Perhaps a `403` response would be more apt here, instead of a 'Not Found' error?
